holoviews/core/data/__init__.py

```diff
@@ -890,6 +952,70 @@ def load_subset(*args):
         return self.interface.groupby(self, dim_names, container_type,
                                       group_type, **kwargs)
 
+    def transform(
+        self,
+        signature=None,
+        dim_transform=None,
+        drop=False,
+        drop_duplicate_data=True,
+        **kwargs
+    ):
+        """
+        Transforms the Dataset according to a dimension transform.
+
+        Args:
+            signature: Specify output arguments as a list of strings
```

Comment:
I'm having a tough time understanding the signature argument. I'm guessing it allows a single transform which returns multiple dicts but I'd like to come up with a better API here, e.g. instead of:

`obj.transform(('a', 'b'), transform)`

I'd probably allow:

`obj.transform((('a', 'b'), transform))`

and allow supplying multiple such transforms using *args.

Reply:
Seems ok to me.

Reply:
Your suggestion looks good!